client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelPool.java

```diff
@@ -309,6 +310,25 @@ public void close() {
       }
    }
 
+   public void inspectPool() {
+      if (terminated || getConnected() > 0 || getActive() > 0) return;
+
+      ChannelOperation cb;
+      long stamp = lock.writeLock();
+      try {
+         cb = callbacks.pollFirst();
```

Comment:
What about the current callback? Is that somehow notified, unfortunately I can't quite see it.

Reply:
Yes. If fail to create the channel, it is canceled, lines 163 and 184. If the channel is created, then the callback is invoked, line 169.